bin/ch/Helpers.cpp

```diff
@@ -16,244 +16,92 @@
-    Helpers::TTReportLastIOErrorAsNeeded(srcByteLength <= dstRemainingByteLength, "The source string must be able to fit within the destination buffer");
-
-    memcpy_s(dst + dpos, dstRemainingByteLength, src, srcByteLength);
-    dst[dpos + srcLength] = _u('\0');
-}
-
-void TTDHostAppendWChar(TTDHostCharType* dst, size_t dstLength, const wchar* src)
+void TTDHostBuildCurrentExeDirectory(char* path, size_t* pathLength)
 {
-    size_t srcLength = wcslen(src);
-    size_t dpos = TTDHostStringLength(dst);
-    Helpers::TTReportLastIOErrorAsNeeded(dpos < dstLength, "The end of the string already exceeds the buffer");
-
-    size_t dstRemainingLength = dstLength - dpos - 1;
-    Helpers::TTReportLastIOErrorAsNeeded(srcLength <= dstRemainingLength, "The source string must be able to fit within the destination buffer");
+    wchar exePath[MAX_PATH];
+    GetModuleFileName(NULL, exePath, MAX_PATH);
 
-    for(size_t i = 0; i < srcLength; ++i)
+    size_t i = wcslen(exePath) - 1;
+    while(exePath[i] != _u('\\'))
     {
-        dst[dpos + i] = (char16)src[i];
+        --i;
     }
 
-    dst[dpos + srcLength] = _u('\0');
-}
-
-void TTDHostAppendAscii(TTDHostCharType* dst, size_t dstLength, const char* src)
-{
-    size_t srcLength = strlen(src);
-    size_t dpos = TTDHostStringLength(dst);
-    Helpers::TTReportLastIOErrorAsNeeded(dpos < dstLength, "The end of the string already exceeds the buffer");
-
-    size_t dstRemainingLength = dstLength - dpos - 1;
-    Helpers::TTReportLastIOErrorAsNeeded(srcLength <= dstRemainingLength, "The source string must be able to fit within the destination buffer");
-
-    for(size_t i = 0; i < srcLength; ++i)
+    if(i * 3 > MAX_URI_LENGTH)
     {
-        dst[dpos + i] = (char16)src[i];
+        printf("Don't overflow path buffer during conversion");
+        exit(1);
     }
-
-    dst[dpos + srcLength] = _u('\0');
+    *pathLength = utf8::EncodeInto((LPUTF8)path, exePath, (charcount_t)(i + 1));
+    path[*pathLength] = '\0';
 }
 
-void TTDHostBuildCurrentExeDirectory(TTDHostCharType* path, size_t pathBufferLength)
+int TTDHostMKDir(const char* path, size_t pathLength)
 {
-    wchar exePath[MAX_PATH];
-    GetModuleFileName(NULL, exePath, MAX_PATH);
+    char16 cpath[MAX_PATH];
+    LPCUTF8 pathbase = (LPCUTF8)path;
+    utf8::DecodeUnitsIntoAndNullTerminate(cpath, pathbase, pathbase + pathLength);
 
-    size_t i = wcslen(exePath) - 1;
-    while(exePath[i] != TTDHostPathSeparatorChar)
-    {
-        --i;
-    }
-    exePath[i + 1] = _u('\0');
-
-    TTDHostAppendWChar(path, pathBufferLength, exePath);
+    return _wmkdir(cpath);
 }
 
-JsTTDStreamHandle TTDHostOpen(const TTDHostCharType* path, bool isWrite)
+JsTTDStreamHandle TTDHostOpen(size_t pathLength, const char* path, bool isWrite)
 {
+    char16 wpath[MAX_PATH];
+    LPCUTF8 pathbase = (LPCUTF8)path;
+    utf8::DecodeUnitsIntoAndNullTerminate(wpath, pathbase, pathbase + pathLength);
```

Comment:
Here too, it seems like these UTF8 methods should take in an output buffer length to be able to ensure that the buffer isn't overrun.